File: decine_ruby/lib/premio.rb
# encoding: UTF-8
# 
#   Fundamentos de Ingenieria del Software
#   Grado en Ingeniería Informática
#
#   2014 © Copyleft - All Wrongs Reserved
# 
#   Ernesto Serrano <erseco@correo.ugr.es>
#   
#   
#

module Decine
  class Premio
    def initialize(premio, categoria, año)
      @premio = premio
      @categoria = categoria
      @año = año
    end
    
    def obtener_datos
      
      resultado = ""
      
      resultado += "\n\tNombre: " + @premio.nombre
      resultado += "\n\tCategoria: " + @categoria
      resultado += "\n\tAño: " + @año.to_s
      resultado += "\n\t------------"

      return resultado
      
    end
    
    
  end
end

File: decine_java/src/decine/Premio.java
/*  
 *  Fundamentos de Ingenieria del Software
 *  Grado en Ingeniería Informática
 * 
 *  2014 © Copyleft - All Wrongs Reserved
 *
 *  Ernesto Serrano <info@ernesto.es>
 *  
 *  
 * 
 */
package decine;

class Premio {
    private FestivalPremio premio;
    private String categoriaPremio;
    private int año;
    
    Premio(FestivalPremio premio, String categoria, int año){
        this.premio = premio;
        this.categoriaPremio = categoria;
        this.año = año;
    }
    
    String obtenerDatos() {
        
        String resultado = "";
        
        resultado += "\tNombre: " + this.premio.getNombre() + "\n";
        resultado += "\tCategoria: " + this.categoriaPremio + "\n";
        resultado += "\tAño: " + String.valueOf(this.año) + "\n";
        resultado += "\t------------\n";
        
        return resultado;
        
    }
}
